POKITTO_LIBS/UIWidgets/UIWidget.cpp

```diff
@@ -0,0 +1,336 @@
+    }
+
+    // Set rect.
+    const int32_t dlgW = Display::getWidth()*2/3;
+    const int32_t dlgH = 11 * fontH;
+    setRect((Display::getWidth()- dlgW) / 2, (Display::getHeight() - dlgH) / 2, dlgW, dlgH);
+
+}
+
+void CancelDlg::draw() {
+
+    update();
+
+    WidgetBase::draw();
+
+    uint16_t currColor = Display::color;
+    Display::color = LB_ITEM_COLOR;
+
+    // Draw little Pokitto!
+    int16_t vx, vy, vw, vh;
+    getViewRect(vx, vy, vw, vh);
+    int16_t x = ((vw - (4*fontW)) / 2); // centralize
+    int16_t y = 0;
+    Display::print(vx + x,vy + y,"ijkl");
+    y+=fontH;
+    Display::print(vx + x,vy + y,"mnop");
+    y+=fontH;
+    Display::print(vx + x,vy + y,"qrst");
+    y+=fontH;
+    Display::print(vx + x,vy + y,"uvwx");
+    y+=fontH;
+
+    // One empty line
+    y+=fontH;
+
+    // Draw text
+    int16_t len = strlen(m_ptext);
+    x = ((vw - (len*fontW)) / 2); // centralize
+    Display::print(vx + x, vy +  y, m_ptext);
+
+    // One empty line
+    y+=fontH;
+
+    // Draw OK button
+    Display::color = m_isOkSelected ? LB_SELECTED_ITEM_COLOR : LB_ITEM_COLOR;
+    len = strlen(m_okText);
+    x = ((vw - (len*fontW)) / 2); // centralize
+    y+=fontH;
+    Display::print(vx + x, vy  + y, m_okText);
+
+    // Draw CANCEL button
+    Display::color = (!m_isOkSelected) ? LB_SELECTED_ITEM_COLOR : LB_ITEM_COLOR;
+    len = strlen(m_cancelText);
+    x = ((vw - (len*fontW)) / 2); // centralize
+    y+=fontH;
+    Display::print(vx + x, vy  + y, m_cancelText);
+
+    // Restore original color.
+    Display::color = currColor;
+}
+
+// Note: the object can be destroyed during this call!
+void CancelDlg::update() {
+
+    if(!m_done) {
+
+        // Done with the dialog?
+        if( Core::buttons.pressed(BTN_A) )
+            m_done = true;
+
+        if (Core::buttons.pressed(BTN_UP) ||
+            Core::buttons.pressed(BTN_DOWN) ||
+            Core::buttons.pressed(BTN_LEFT) ||
+            Core::buttons.pressed(BTN_RIGHT) ) {
+
+            m_isOkSelected = !m_isOkSelected;  // Switch state
+        }
+    }
+}
+
+// ListBoxDlg class
+
+ListBoxDlg::ListBoxDlg(char* titleText, int16_t x, int16_t y, int16_t w, int16_t h, uint16_t maxItems )
+: DialogBase(WidgetBase::hasBorders){
+
+    m_listbox = nullptr;
+
+    init(titleText, x, y, w, h, maxItems);
+}
+
+ListBoxDlg::~ListBoxDlg(){
+
+    delete m_listbox;
+}
+
+void ListBoxDlg::init(char* titleText, int16_t x, int16_t y, int16_t w, int16_t h, uint16_t maxItems ) {
+
+    // Set title
+    if(titleText)
+        setTitle(titleText);
+
+    // Set the dialog rect
+    setRect(x, y, w, h);
+
+    // Create listbox widget.
+    int16_t vx, vy, vw, vh;
+    getViewRect(vx, vy, vw, vh);
+    m_listbox = new(std::nothrow) ListBox(0);
+    m_listbox->init(vx, vy, vw/fontW, vh/fontH + 1, maxItems);
+}
+
+uint16_t ListBoxDlg::addItem(const char* text) {
+    m_listbox->addItem(text);
+}
+
+uint16_t ListBoxDlg::removeItem(uint8_t itemId) {
+    m_listbox->removeItem(itemId);
+}
```

Comment:
These methods have no return. Please change them to void or return something. The compiler already warns about it. Please don't ignore :)